features/telemetry/daemon/src/main/java/org/opennms/netmgt/telemetry/daemon/ConnectorManager.java

```diff
@@ -88,28 +91,43 @@ private void startStreamingFor(ConnectorConfig connectorConfig, PackageConfig pa
             if (connectorsByKey.containsKey(key)) {
                 LOG.debug("Connector already exists. Ignoring.");
             }
-
+            List<Map<String, String>> interpolatedMapList = new ArrayList<>();
             // Flatten the parameters to a map
             Map<String,String> parmMap = packageConfig.getParameters().stream()
+                    .filter(parameter -> Strings.isNullOrEmpty(parameter.getGroup()))
                     .collect(Collectors.toMap(
                             Parameter::getKey,
                             Parameter::getValue
                     ));
             // Interpolate meta-data in parameter values
-            parmMap = Interpolator.interpolateStrings(parmMap, new FallbackScope(
-                    entityScopeProvider.getScopeForNode(serviceRef.getNodeId()),
-                    entityScopeProvider.getScopeForInterface(serviceRef.getNodeId(), InetAddressUtils.str(serviceRef.getIpAddress())),
-                    entityScopeProvider.getScopeForService(serviceRef.getNodeId(), serviceRef.getIpAddress(), serviceRef.getServiceName())
-            ));
+            parmMap = getInterpolated(parmMap, serviceRef);
+            interpolatedMapList.add(parmMap);
+
+            // Interpolate grouped parameters
+            Map<String, Map<String, String>> parmMapByGroup = packageConfig.getParameters().stream()
+                    .filter(parameter -> !Strings.isNullOrEmpty(parameter.getGroup()))
+                    .collect(Collectors.groupingBy(Parameter::getGroup, Collectors.toMap(Parameter::getKey, Parameter::getValue)));
+
+            parmMapByGroup.forEach((group, parmeterMap) -> {
+                interpolatedMapList.add(getInterpolated(parmeterMap, serviceRef));
+            });
```

Comment:
Looks like the whole block just interpolates all keys regardless of the group.

Maybe something like this does the job:
```java
List<Map<String, String>> interpolatedMapList = packageConfig.getParameters().stream()
    .map(parmeterMap -> getInterpolated(parmeterMap, serviceRef))
    .collect(Collectors.toList());
```

Reply:
This is to put parameters that belong to one group into one set of params. Modified this and added test case to show how grouping is done.